packages/repeater/src/lib/Repeater.ts

```diff
@@ -27,20 +36,46 @@ export class Repeater {
     this.repeaterId = repeaterId;
     this.bus = bus;
     this.configuration = configuration;
+
+    const { container } = this.configuration;
+    if (container.isRegistered(Logger, true)) {
+      this.logger = container.resolve(Logger);
+    }
+
+    this.setupShutdown();
   }
 
   public async start(): Promise<void> {
-    const res = await this.register();
-    if (!res) {
-      throw new Error('Error registering repeater.');
+    if (this.runningStatus !== RunningStatus.OFF) {
+      throw new Error('Repeater is already active.');
     }
 
-    await this.subscribeToEvents();
+    this.runningStatus = RunningStatus.STARTING;
+
+    try {
+      const res = await this.register();
+      if (!res) {
+        throw new Error('Error registering repeater.');
+      }
```

Comment:
I missed that when skimming last time) How to handle response on `RegisterRepeaterCommand`, see  https://github.com/NeuraLegion/nexploit-cli/blob/74e7ca3cdde68b85998a6d6f8b8051cb65311418/src/Repeater/DefaultRepeaterLauncher.ts#L146-L168